crates/router/src/db/dispute.rs

```diff
@@ -131,54 +135,739 @@ impl DisputeInterface for Store {
+        use serde_json::Value;
+        use storage_models::{
+            dispute::{Dispute, DisputeNew},
+            enums::{DisputeStage, DisputeStatus},
+        };
+        use time::macros::datetime;
+
+        use crate::db::{dispute::DisputeInterface, MockDb};
+
+        fn disputes_eq(d1: Dispute, d2: Dispute) {
+            assert_eq!(d1.id, d2.id);
+            assert_eq!(d1.dispute_id, d2.dispute_id);
+            assert_eq!(d1.amount, d2.amount);
+            assert_eq!(d1.currency, d2.currency);
+            assert_eq!(d1.dispute_stage, d2.dispute_stage);
+            assert_eq!(d1.dispute_status, d2.dispute_status);
+            assert_eq!(d1.payment_id, d2.payment_id);
+            assert_eq!(d1.attempt_id, d2.attempt_id);
+            assert_eq!(d1.merchant_id, d2.merchant_id);
+            assert_eq!(d1.connector_status, d2.connector_status);
+            assert_eq!(d1.connector_dispute_id, d2.connector_dispute_id);
+            assert_eq!(d1.connector_reason, d2.connector_reason);
+            assert_eq!(d1.connector_reason_code, d2.connector_reason_code);
+            assert_eq!(d1.challenge_required_by, d2.challenge_required_by);
+            assert_eq!(d1.connector_created_at, d2.connector_created_at);
+            assert_eq!(d1.connector_updated_at, d2.connector_updated_at);
+            assert_eq!(d1.created_at, d2.created_at);
+            assert_eq!(d1.modified_at, d2.modified_at);
+            assert_eq!(d1.connector, d2.connector);
+            assert_eq!(d1.evidence, d2.evidence);
+        }
+
+        pub struct DisputeNewIds {
+            dispute_id: String,
+            payment_id: String,
+            attempt_id: String,
+            merchant_id: String,
+            connector_dispute_id: String,
+        }
+
+        fn create_dispute_new(dispute_ids: DisputeNewIds) -> DisputeNew {
+            DisputeNew {
+                dispute_id: dispute_ids.dispute_id,
+                amount: "amount".into(),
+                currency: "currency".into(),
+                dispute_stage: DisputeStage::Dispute,
+                dispute_status: DisputeStatus::DisputeOpened,
+                payment_id: dispute_ids.payment_id,
+                attempt_id: dispute_ids.attempt_id,
+                merchant_id: dispute_ids.merchant_id,
+                connector_status: "connector_status".into(),
+                connector_dispute_id: dispute_ids.connector_dispute_id,
+                connector_reason: Some("connector_reason".into()),
+                connector_reason_code: Some("connector_reason_code".into()),
+                challenge_required_by: Some(datetime!(2019-01-01 0:00)),
+                connector_created_at: Some(datetime!(2019-01-02 0:00)),
+                connector_updated_at: Some(datetime!(2019-01-03 0:00)),
+                connector: "connector".into(),
+                evidence: Some(Secret::from(Value::String("evidence".into()))),
+            }
+        }
+
+        #[allow(clippy::unwrap_used)]
```

Comment:
Optional change: you can have a `#![allow(clippy::unwrap_used)]` at the top, within the `tests` module.